includes/classes/SiteAutomation/EndPoints.php

```diff
@@ -0,0 +1,343 @@
+
+		$rules = [
+			'display_featured_image' => $display_featured_image,
+			'display_author'         => $display_author,
+			'display_post_date'      => $display_post_date,
+			'display_post_excerpt'   => $display_post_excerpt,
+		];
+
+		// If override ID exists, only return the post data.
+		// This is required only when updating (add/replace) the innerBlocks.
+		if ( ! empty( $override_post_ID ) ) {
+			$curated_posts     = [];
+			$post_data         = get_post( $override_post_ID );
+			$post_data_updated = $this->get_post_details( $override_post_ID, $rules );
+			$curated_posts[]   = (object) array_merge( (array) $post_data, (array) $post_data_updated );
+
+			return $curated_posts;
+		}
+
+		foreach ( $curated_posts as $index => $curated_post ) {
+			$post_data_updated       = $this->get_post_details( $curated_post->ID, $rules );
+			$curated_posts[ $index ] = (object) array_merge( (array) $curated_post, (array) $post_data_updated );
+		}
+
+		return $curated_posts;
+	}
+
+	/**
+	 * Get the post data with prepared items to be rendered to the inner blocks.
+	 *
+	 * @param int   $post_ID Post ID.
+	 * @param array $rules array of the parent block settings.
+	 *
+	 * @return \stdClass Prepared post data.
+	 */
+	public function get_post_details( $post_ID, $rules ) {
+		$post_data = new \stdClass();
+
+		$post_data->postLink = get_the_permalink( $post_ID );
+		if ( $rules['display_featured_image'] ) {
+			$post_data->featuredImage = get_the_post_thumbnail( $post_ID );
+		}
+		if ( $rules['display_author'] ) {
+			$author_id             = get_post_field( 'post_author', $post_ID );
+			$author_display_name   = get_the_author_meta( 'display_name', $author_id );
+			$byline                = sprintf( __( 'by %s', 'sophi-wp' ), $author_display_name );
+			$post_data->postAuthor = $byline;
+		}
+		if ( $rules['display_post_date'] ) {
+			$post_data->postDate  = get_the_date( '', $post_ID );
+			$post_data->postDateC = get_the_date( 'c', $post_ID );
+		}
+		if ( $rules['display_post_excerpt'] ) {
+			$post_data->post_excerpt = wp_kses_post( get_the_excerpt( $post_ID ) );
+		}
+
+		return $post_data;
+	}
+
+	/**
+	 * Get the query params to get the sophi posts.
+	 *
+	 * @since 1.3.0
+	 *
+	 * @return array
+	 */
+	public function site_automation_params() {
+		$params['pageName'] = array(
+			'description'       => __( 'Name of the page.', 'sophi-wp' ),
+			'required'          => true,
+			'type'              => 'string',
+			'sanitize_callback' => 'sanitize_text_field',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+
+		$params['widgetName'] = array(
+			'description'       => __( 'Name of the widget.', 'sophi-wp' ),
+			'required'          => true,
+			'type'              => 'string',
+			'sanitize_callback' => 'sanitize_text_field',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+
+		/**
+		 * Filters the get sophi posts query params.
+		 *
+		 * @param array $params Query params.
+		 *
+		 * @since 1.3.0
+		 */
```

Comment:
This docblock needs a few changes in order for this to show up in our automated hook docs. See example [here](https://github.com/globeandmail/sophi-for-wordpress/blob/develop/includes/classes/SiteAutomation/Request.php#L77). Main thing to note is the use of `@hook` and the brackets around the params.

Reply:
https://github.com/globeandmail/sophi-for-wordpress/pull/333/commits/d7f3bb5f76704d21c9cf317832c74ef2d0842ca3